src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -1290,6 +1299,7 @@ public PropertyFetch(Type? type)
                         Justification = "In EventSource, EnsureDescriptorsInitialized's use of GetType preserves this method which " +
                                         "requires unreferenced code, but EnsureDescriptorsInitialized does not access this member and is safe to call.")]
                     [RequiresUnreferencedCode(DiagnosticSource.WriteRequiresUnreferencedCode)]
+                    [RequiresDynamicCode(DiagnosticSource.WriteRequiresUnreferencedCode)]
```

Comment:
I assume that this warning is due to `MakeGenericType` below. I do not understand how this message justifies that it is safe for AOT.